pom.client.xml

```diff
@@ -647,30 +640,59 @@
                 <arg>-Xlint:-deprecation</arg>
                 <arg>-Xlint:-processing</arg>
               </compilerArgs>
+              <excludes>
+                <exclude>module-info.java</exclude>
+              </excludes>
+            </configuration>
+          </plugin>
+          <!-- Avoid errors from module-info -->
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-javadoc-plugin</artifactId>
+            <version>${maven-javadoc.version}</version>
+            <configuration>
+              <sourceFileExcludes>
+                <sourceFileExclude>module-info.java</sourceFileExclude>
+              </sourceFileExcludes>
             </configuration>
           </plugin>
         </plugins>
       </build>
+      <reporting>
+        <plugins>
+          <!-- Avoid errors from module-info -->
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-javadoc-plugin</artifactId>
+            <version>${maven-javadoc.version}</version>
+            <configuration>
+              <sourceFileExcludes>
+                <sourceFileExclude>module-info.java</sourceFileExclude>
+              </sourceFileExcludes>
+            </configuration>
+          </plugin>
+        </plugins>
+      </reporting>
     </profile>
 
+    <!-- Setup for Java 9+ -->
     <profile>
-      <id>java-lts</id>
+      <id>java9plus</id>
       <activation>
-        <property>
-          <name>java-lts</name>
-        </property>
+        <jdk>[9,)</jdk>
       </activation>
       <build>
         <plugins>
           <plugin>
             <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-compiler-plugin</artifactId>
             <version>${maven-compiler-plugin.version}</version>
+            <!-- setup defaults for compile and testCompile -->
             <configuration>
-              <source>11</source>
-              <target>11</target>
+              <source>9</source>
+              <target>9</target>
               <showWarnings>true</showWarnings>
-              <failOnWarning>true</failOnWarning>
+              <failOnWarning>false</failOnWarning> <!-- FIXME: this is false for now as it introduces a build failure -->
```

Comment:
Do we still have build failures if this is true?

Reply:
Yes, the build will still fail because we are exporting some implementation packages to client library modules and a clean build will not know in advance all these modules and logs warning about "<modulename> is not known"

Reply:
Please file an issue. Of all the issues to be filed, this is the one I want resolved most urgently. Do you have any plans on how to resolve this?